evaluate/task/task-transpile.go

```diff
@@ -0,0 +1,178 @@
+	// Check if the package path has a directory called "implementation".
+	var implementationPath string
+	packageFiles, err := os.ReadDir(packagePath)
+	if err != nil {
+		return "", "", pkgerrors.WithStack(err)
+	}
+	for _, file := range packageFiles {
+		if file.IsDir() && file.Name() == "implementation" {
+			implementationPath = filepath.Join(packagePath, file.Name())
+		}
+	}
+	if implementationPath == "" {
+		return "", "", pkgerrors.Errorf("package %q in repository %q must have an \"implementation\" directory with a %s source file to transpile", packagePath, ctx.Repository.Name(), ctx.Language.Name())
+	}
+
+	// Check if the "implementation" directory contains only one source file in the language being transpiled from.
+	implementationFiles, err := os.ReadDir(implementationPath)
+	if err != nil {
+		return "", "", pkgerrors.WithStack(err)
+	} else if len(implementationFiles) != 1 {
+		return "", "", pkgerrors.Errorf("package %q in repository %q must have an \"implementation\" directory with just one %s source file to transpile", packagePath, ctx.Repository.Name(), ctx.Language.Name())
+	}
+	sourceFilePath = filepath.Join(implementationPath, implementationFiles[0].Name())
+	if filepath.Ext(sourceFilePath) != ctx.Language.DefaultFileExtension() {
+		return "", "", pkgerrors.Errorf("package %q in repository %q must have an \"implementation\" directory with a %s source file", packagePath, ctx.Repository.Name(), ctx.Language.Name())
+	} else if strings.HasPrefix(sourceFilePath, ctx.Language.DefaultTestFileSuffix()) {
+		return "", "", pkgerrors.Errorf("package %q in repository %q must have an \"implementation\" directory with only a %s source file, but found a test file %q", packagePath, ctx.Repository.Name(), ctx.Language.Name(), sourceFilePath)
+	}
+
+	// Check if the package path contains a source file and the corresponding test file in the target language for transpilation.
+	var hasTestFile bool
+	files, err := targetLanguage.Files(fileLogger, packagePath)
+	if err != nil {
+		return "", "", pkgerrors.WithStack(err)
+	} else if len(files) != 2 {
+		return "", "", pkgerrors.Errorf("package %q in repository %q must have a %s source file and the corresponding test file", packagePath, ctx.Repository.Name(), targetLanguage.Name())
+	}
+	for _, file := range files {
+		if strings.HasSuffix(file, targetLanguage.DefaultTestFileSuffix()) {
+			hasTestFile = true
+		} else if filepath.Ext(file) == targetLanguage.DefaultFileExtension() {
+			stubFilePath = filepath.Join(packagePath, file)
+		}
+	}
+	if !hasTestFile {
+		return "", "", pkgerrors.Errorf("package %q in repository %q must have a %s test file", packagePath, ctx.Repository.Name(), targetLanguage.Name())
+	} else if stubFilePath == "" {
+		return "", "", pkgerrors.Errorf("package %q in repository %q must have a %s file with a stub function", packagePath, ctx.Repository.Name(), targetLanguage.Name())
+	}
```

Comment:
We have this check already for the code repair, right? Maybe we can share something :thinking: